File: java/src/main/java/mx/iteso/privileges/Delete.java
package mx.iteso.privileges;

import mx.iteso.PrivilegeDecorator;
import mx.iteso.User;

/**
 * Delete class.
 */
public class Delete extends PrivilegeDecorator {

    /**
     * User.
     */
    private final User user;

    /**
     * Constructor.
     *
     * @param user user.
     */
    public Delete(final User user) {
        this.user = user;
    }

    /**
     * Get Description.
     *
     * @return String.
     */
    @Override
    public String getDescription() {
        return "Delete permission to: " + user.getDescription();
    }

    /**
     * Publish Permission.
     *
     * @return boolean.
     */
    public boolean publishPermission() {
        return user.publishPermission();
    }

    /**
     * Delete Permission.
     *
     * @return boolean.
     */
    public boolean deletePermission() {
        System.out.println("Delete permission added");
        return true;
    }

    /**
     * Approve permission.
     *
     * @return boolean.
     */
    public boolean approvePermission() {
        return user.approvePermission();
    }

    /**
     * Comment Permission.
     *
     * @return boolean.
     */
    public boolean commentPermission() {
        return user.commentPermission();
    }
}

File: python/src/main/python/Delete.py
class Delete():

    def __init__(self, user):
        self.user = user

    def getDescription(self):
        return "Delete permission to: " + self.user.getDescription()

    def publishPermission(self):
        return self.user.publishPermission()

    def deletePermission(self):
        print("Delete permission added")
        return True

    def approvePermission(self):
        return self.user.approvePermission()

    def commentPermission(self):
        return self.user.commentPermission()
